core/src/main/scala/kafka/log/AbstractIndex.scala

```diff
@@ -374,7 +374,7 @@ abstract class AbstractIndex[K, V](@volatile var file: File, val baseOffset: Lon
       var lo = begin
       var hi = end
       while(lo < hi) {
-        val mid = ceil(hi/2.0 + lo/2.0).toInt
+        val mid = lo + (hi - lo) / 2
```

Comment:
Is there a reason why we can't do the same as Java's binary search? That is:

```java
int mid = (low + high) >>> 1;
```

Reply:
Explanation in https://ai.googleblog.com/2006/06/extra-extra-read-all-about-it-nearly.html

Reply:
Reference for the above logic is here [DualPivotQuicksort](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/tip/src/share/classes/java/util/DualPivotQuicksort.java#l289)

Reply:
Bump @cmccabe